### PR TITLE
feat(builder): use builtin:swc-loader 

### PR DESCRIPTION
**Pull request by 9aoy**
### <samp>🤖 Generated by Copilot at 5315fe6</samp>

This pull request refactors the builder-rspack-provider package and its tests to use the swc-loader plugin instead of the swc plugin for minimizing and transpiling JavaScript code with swc. The swc-loader plugin provides some default and customized swc configurations for rspack experiments, such as react and polyfill. The pull request also updates the dependencies of the package to use the canary versions of rspack packages based on swc.

### <samp>🤖 Generated by Copilot at 5315fe6</samp>

*  Update dependencies of `builder-rspack-provider` to use canary versions of swc-based packages ([link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-0dfad59d3b9faa8b69b516527ff7dcc43db019040e3499dfa12a7b716d2e7ab7L59-R61))
*  Simplify output config schema to remove polyfill option ([link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-e524d13b34378b73e0c008099b11dbb5f1a565d43e8d73f456349d1c4a306995L5-R5))
*  Rename import and use of SwcJsMinimizerPlugin to SwcJsMinimizerRspackPlugin, which supports rspack experiments options ([link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-00652a3b9cfcf087a07f68d8d17fcba9f6529ef9ad06eff43a5d63bb7be474f3L8-R8), [link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-00652a3b9cfcf087a07f68d8d17fcba9f6529ef9ad06eff43a5d63bb7be474f3L45-R45))
*  Modify `builderPluginReact` to use `swc-loader` plugin and set rspack experiments options for react on it ([link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-c13ca41509e796836a9c3413111a005735197d8956d381156301a1e88abf8046L7-R28))
*  Add `swc-loader` plugin, which provides default and customized swc configurations and handles polyfill entry mode for web targets ([link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-452962aa6baee69bb6664bcf889397df84a1bfa868476fc8fdb4c083b819c551R1-R165))
*  Update default plugins to use `swc-loader` plugin and comment out `swc` plugin ([link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-9c09796bb9593836c398dbbe112387c94c45180426120aa80634fad89cd9cd23L40-R42))
*  Modify `isUsingHMR` function to only accept isProd and target properties ([link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-e8a82071101f00fb2bf3fde8c73e67dde08ad1da5198c5f123848cded3dfff8cL5-R5))
*  Update polyfill test case to import helper function, remove webpackOnlyTest wrapper, add runServer option, and use playwright page object to verify polyfill behavior ([link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-e6a56ef315e0230b0c5ebe91866ce96d0143fbafe4cf5653ee9f886d75505767L3-R3), [link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-e6a56ef315e0230b0c5ebe91866ce96d0143fbafe4cf5653ee9f886d75505767L41-R40), [link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-e6a56ef315e0230b0c5ebe91866ce96d0143fbafe4cf5653ee9f886d75505767L51-R55))
*  Modify polyfill test source file to assign flat array result to window property ([link](https://github.com/web-infra-dev/modern.js/pull/4626/files?diff=unified&w=0#diff-ddf19e68aa0a48795f71855ae00c526a160f44316bd930443858b62be6df25e9L1-R2))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder-rspack-provider/src/shared/rspackVersion.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export const getRspackVersion = async (): Promise<string> => {
 };
 
 // depend on rspack builtin plugins
-export const supportedRspackMinimumVersion = '0.3.3';
+export const supportedRspackMinimumVersion = '0.3.4';
```
Comment (chenjiahan) [inline packages/builder/builder-rspack-provider/src/shared/rspackVersion.ts:17]: 0.3.5?

File: packages/builder/builder-rspack-provider/src/types/rspack.ts [hunk 2]
```diff
@@ -6,15 +6,113 @@ type MultiCompiler = Rspack.MultiCompiler;
...
+    import?: BuiltinsOptions['pluginImport'];
+  };
+};
+
 export interface RspackConfig extends RspackOptions {
   /** multi type is useless in builder and make get value difficult */
   entry?: Record<string, string | string[]>;
-  // can't use htmlPlugin & builtins.html at the same time.
+  /**
+   * `builtins.react / pluginImport / decorator / presetEnv / emotion / relay` not support in rspack-provider,
```
Comment (chenjiahan) [inline packages/builder/builder-rspack-provider/src/types/rspack.ts:111]: not support in rspsck-provider -> are no longer supported by Rspsck

File: packages/document/builder-doc/docs/en/guide/advanced/rspack-start.mdx [hunk 3]
```diff
@@ -196,9 +195,23 @@ When using Rspack for building, you can still configure Babel plugins through [t
 
 ### 5. SWC Configuration Support
 
-Rspack has built-in support for some swc configurations, such as builtins.react, which can be referred to in [Builtins](https://www.rspack.dev/config/builtins.html).
+Builder uses Rspack [builtin:swc-loader](https://www.rspack.dev/guide/loader.html#builtinswc-loader) for code translation. If you have custom configuration requirements for builtin:swc-loader, you can Refer to the following code:
```
Comment (chenjiahan) [inline packages/document/builder-doc/docs/en/guide/advanced/rspack-start.mdx:198]: Refer -> refer

File: packages/builder/builder-rspack-provider/src/plugins/swc.ts [hunk 4]
```diff
@@ -1,20 +1,49 @@
...
+      },
+      // Avoid the webpack magic comment to be removed
+      // https://github.com/swc-project/swc/issues/6403
+      preserveAllComments: true,
+    },
+    isModule: 'unknown',
+    minify: false, // for loader, we don't need to minify, we do minification using plugin
+    sourceMaps: true,
+    env: {
+      targets: '> 0.01%, not dead, not op_mini all',
```
Comment (chenjiahan) [inline packages/builder/builder-rspack-provider/src/plugins/swc.ts:41]: Should we use the browserslist config here?

File: packages/builder/builder-rspack-provider/src/plugins/swc.ts [hunk 5]
```diff
@@ -1,20 +1,49 @@
...
+        syntax: 'typescript',
+        decorators: true,
+      },
+      // Avoid the webpack magic comment to be removed
+      // https://github.com/swc-project/swc/issues/6403
+      preserveAllComments: true,
+    },
+    isModule: 'unknown',
+    minify: false, // for loader, we don't need to minify, we do minification using plugin
+    sourceMaps: true,
```
Comment (chenjiahan) [inline packages/builder/builder-rspack-provider/src/plugins/swc.ts:39]: How about the disableSourceMap option

File: packages/builder/builder-rspack-provider/src/plugins/swc.ts (hunk 4 above)
Comment (9aoy) [reply packages/builder/builder-rspack-provider/src/plugins/swc.ts]: done

File: packages/document/builder-doc/docs/en/guide/advanced/rspack-start.mdx (hunk 3 above)
Comment (9aoy) [reply packages/document/builder-doc/docs/en/guide/advanced/rspack-start.mdx]: done

File: packages/builder/builder-rspack-provider/src/types/rspack.ts (hunk 2 above)
Comment (9aoy) [reply packages/builder/builder-rspack-provider/src/types/rspack.ts]: done

File: packages/builder/builder-rspack-provider/src/shared/rspackVersion.ts (hunk 1 above)
Comment (9aoy) [reply packages/builder/builder-rspack-provider/src/shared/rspackVersion.ts]: done

File: packages/builder/builder-rspack-provider/src/plugins/swc.ts (hunk 5 above)
Comment (9aoy) [reply packages/builder/builder-rspack-provider/src/plugins/swc.ts]: done

**Pull request merged**